### PR TITLE
Adding parseInt to config.timeout

### PR DESCRIPTION
**Pull request by piiih**
Issue: https://github.com/axios/axios/issues/3778

### CODE REVIEW

File: lib/adapters/http.js [hunk 1]
```diff
@@ -279,15 +279,29 @@ module.exports = function httpAdapter(config) {
 
     // Handle request timeout
     if (config.timeout) {
+      // This is forcing a int timeout to avoid problems if the `req` interface doesn't handle other types.
+      var timeout = parseInt(config.timeout, 10);
```
Comment (jasonsaayman) [inline lib/adapters/http.js:283]: Can we please change this to `let timeout=`

File: test/unit/adapters/http.js [hunk 2]
```diff
@@ -29,6 +29,64 @@ describe('supports http with nodejs', function () {
     }
   });
 
+  it('should NOT parse the timeout property', function (done) {
```
Comment (jasonsaayman) [inline test/unit/adapters/http.js:32]: Can this test rather be something like `should throw an error if the timeout property is not parsable as a number`

**Review by jasonsaayman — COMMENTED** [review]
Please check issues

File: lib/adapters/http.js (hunk 1 above)
Comment (piiih) [reply lib/adapters/http.js:283]: I tryed this, but lint doesn't allow this :/

Comment (jasonsaayman) [reply lib/adapters/http.js:283]: Forgot no es6 so that's fine leave it as is.

File: test/unit/adapters/http.js (hunk 2 above)
Comment (piiih) [reply test/unit/adapters/http.js]: Done! :)

**Review by jasonsaayman — APPROVED** [review]
👍

**Pull request merged**